docs/docs/python/integrations/aws-lambda.mdx

```diff
@@ -69,7 +69,7 @@ Use [layout](../../../reference/targets/python_aws_lambda_function.mdx#layout) t
 
 When building an Lambda artifact, Pants and the underlying Pex tool need to know details about target runtime to be able to choose appropriate artifacts for third-party dependencies that have native code. These details can be inferred or provided in three ways, from highest precedence to lowest precedence:
 
-1. An explicit value for [the `complete_platforms` field](../../../reference/targets/python_aws_lambda_function.mdx#complete_platforms). The "complete platforms" are the underlying source of truth.
+1. An explicit value for [the `complete_platforms` field](../../../reference/targets/python_aws_lambda_function.mdx#complete_platforms). The "complete platforms" are the underlying source of truth. For instructions on how to generate these files for Lambda, refer to [Update or create FaaS platforms files](../../contributions/development/maintenance-tasks-and-scripts.mdx#aws-lambda).
```

Comment:
I appreciate tweaks here, but I think we could improve the docs here some more:

- Some of this content applies to GCF too and so could be added to that file in a similar way (duplication is unfortunate, but tolerable, I think, for this sort of docs, especially given there's a few subtle differences between the platforms)
- I don't think a user should be encouraged to use the internal Pants script if they need to provide an explicit value for the `complete_platforms`: 
   - to use it, they'll need to build Pants itself, run the script for _all_ versions and then copy out the appropriate update one for them, and that sounds like a lot of overhead compared to just finding the right docker image(s) for their specific circumstances and running that directly. 
   - Instead I suggest linking to the new PEX docs here, And rephrasing the first sentence of the "Alternatively" section to be something like "Alternatively, you can generate the `complete_platforms` file by running a cloud function that outputs the appropriate file contents.")
- Very briefly linking to the internal script docs in item 2 (where it talks about the built-in files) seems sensible, to give an interested person breadcrumbs to follow, but even there we shouldn't be emphasising it: users mostly shouldn't need to care about the details of how the files are generated! For example, it could just be turning "static exports" into a link with no further details.


(Btw, it's feeling like we're going in circles a bit here, so I'm concerned that I'm communicating poorly, please let me know if things aren't making sense.)

Reply:
I don't think I fully understand the intention behind your comment here.

I get this part:

> I don't think a user should be encouraged to use the internal Pants script if they need to provide an explicit value for the complete_platforms:
>
> to use it, they'll need to build Pants itself, run the script for all versions and then copy out the appropriate update one for them, and that sounds like a lot of overhead compared to just finding the right docker image(s) for their specific circumstances and running that directly.

But not this part:

> Instead I suggest linking to the new PEX docs here, And rephrasing the first sentence of the "Alternatively" section to be something like "Alternatively, you can generate the complete_platforms file by running a cloud function that outputs the appropriate file contents.")

The PEX docs already link out to both AWS Lambda and GCF docs per the other comment thread, and I'm not quite sure how adding a link the other way helps? Instead, what I did was include the cloud function inline.

> Very briefly linking to the internal script docs in item 2 (where it talks about the built-in files) seems sensible, to give an interested person breadcrumbs to follow, but even there we shouldn't be emphasising it: users mostly shouldn't need to care about the details of how the files are generated! For example, it could just be turning "static exports" into a link with no further details.

This is fair feedback, I removed links to the internal script docs to avoid confusion. I suppose we could add those links back somewhere within these docs if desired, but if we're looking to de-emphasize them then I don't see the point of including the links at all.